File: app/src/main/java/com/slapshotapps/dragonshockey/dialogs/StatsSortDialogFragment.java
package com.slapshotapps.dragonshockey.dialogs;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;
import androidx.appcompat.app.AlertDialog;

public class StatsSortDialogFragment extends DialogFragment {

    public interface StatsSortSelectionListener {
        void onSortOptionSelected(StatSortSelection sortSelection);
    }

    private static final String CURRENT_SORT_OPTION_KEY = "currentSortOption";

    private final StatSortSelection[] statSortOptions = {
        StatSortSelection.GamesPlayed, StatSortSelection.Goals, StatSortSelection.Assists,
        StatSortSelection.Points, StatSortSelection.PenaltyMinutes
    };

    private StatsSortSelectionListener listener;

    public static StatsSortDialogFragment newInstance(StatSortSelection currentSortSelection) {
        Bundle args = new Bundle();
        StatsSortDialogFragment fragment = new StatsSortDialogFragment();

        args.putSerializable(CURRENT_SORT_OPTION_KEY, currentSortSelection);
        fragment.setArguments(args);
        return fragment;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        StatSortSelection currentSelection =
            (StatSortSelection) getArguments().getSerializable(CURRENT_SORT_OPTION_KEY);

        final String[] items = {
            statSortOptions[0].getStatName(), statSortOptions[1].getStatName(),
            statSortOptions[2].getStatName(), statSortOptions[3].getStatName(), statSortOptions[4].getStatName()
        };

        AlertDialog.Builder builder = new AlertDialog.Builder(getContext()).setTitle("Stat Sorting")
            .setSingleChoiceItems(items,
                currentSelection != null ? currentSelection.getIndex()
                    : StatSortSelection.Points.getIndex(),
                new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        if (listener != null) {
                            listener.onSortOptionSelected(statSortOptions[which]);
                            dismiss();
                        }
                    }
                });

        return builder.create();
    }

    public void setListener(StatsSortSelectionListener listener) {
        this.listener = listener;
    }
}

File: app/src/main/java/com/slapshotapps/dragonshockey/dialogs/StatsSortDialogFragment.kt
package com.slapshotapps.dragonshockey.dialogs

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment

class StatsSortDialogFragment : DialogFragment() {

  private val statSortOptions = arrayOf(StatSortSelection.GamesPlayed, StatSortSelection.Goals,
      StatSortSelection.Assists, StatSortSelection.Points, StatSortSelection.PenaltyMinutes)

  private var listener: StatsSortSelectionListener? = null

  interface StatsSortSelectionListener {
    fun onSortOptionSelected(sortSelection: StatSortSelection)
  }

  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

    val currentSelection = arguments!!.getSerializable(CURRENT_SORT_OPTION_KEY) as StatSortSelection

    val items = arrayOf(statSortOptions[0].statName, statSortOptions[1].statName,
        statSortOptions[2].statName, statSortOptions[3].statName, statSortOptions[4].statName)

    val builder = AlertDialog.Builder(context!!).setTitle("Stat Sorting")
        .setSingleChoiceItems(items, currentSelection.index) { _, which ->
          if (listener != null) {
            listener!!.onSortOptionSelected(statSortOptions[which])
            dismiss()
          }
        }

    return builder.create()
  }

  fun setListener(listener: StatsSortSelectionListener) {
    this.listener = listener
  }

  companion object {

    private val CURRENT_SORT_OPTION_KEY = "currentSortOption"

    fun newInstance(currentSortSelection: StatSortSelection): StatsSortDialogFragment {
      val args = Bundle()
      val fragment = StatsSortDialogFragment()

      args.putSerializable(CURRENT_SORT_OPTION_KEY, currentSortSelection)
      fragment.arguments = args
      return fragment
    }
  }
}
